extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlPatch.java

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.custom.V1Patch;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+
+public class KubectlPatch<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlPatch<ApiType>>
+    implements Kubectl.Executable<ApiType> {
+
+  private Class<ApiType> apiTypeClass;
+  private Class<KubernetesListObject> apiListTypeClass;
+  private V1Patch patchContent;
+
+  KubectlPatch(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+  }
+
+  public KubectlPatch apiTypeClass(Class<ApiType> apiTypeClass) {
+    this.apiTypeClass = apiTypeClass;
+    return this;
+  }
+
+  public KubectlPatch apiListTypeClass(Class<KubernetesListObject> apiListTypeClass) {
+    this.apiListTypeClass = apiListTypeClass;
+    return this;
+  }
+
+  public KubectlPatch patchContent(V1Patch patchContent) {
+    this.patchContent = patchContent;
+    return this;
+  }
+
```

Comment:
Do we want to add a convenience `patch(ApiType object) { ... }` method that constructs the patch for the person? I think that would be useful.

Reply:
@brendandburns - You mean a method `patch` which return `V1Patch` object for the user?